tests/VirtoCommerce.DemoCustomerSegmentsModule.Tests/CustomerSegmentExpressionTest.cs

```diff
@@ -0,0 +1,358 @@
+                {
+                    new DynamicPropertyObjectValue
+                    {
+                        Locale = "en-US", Value = "Value1", ValueType = DynamicPropertyValueType.ShortText
+                    },
+                    new DynamicPropertyObjectValue
+                    {
+                        Locale = "fr-FR", Value = "Value1", ValueType = DynamicPropertyValueType.ShortText
+                    },
+                    new DynamicPropertyObjectValue
+                    {
+                        Locale = "de-DE", Value = "Value2", ValueType = DynamicPropertyValueType.ShortText
+                    }
+                }
+            };
+            MultiLanguageDynamicPropertyWithMultipleValues =
+                (DynamicObjectProperty) MultiLanguageDynamicProperty.Clone();
+            MultiLanguageDynamicPropertyWithMultipleValues.Values = MultiLanguageDynamicProperty.Values.Concat(new[]
+            {
+                new DynamicPropertyObjectValue
+                {
+                    Locale = "en-US", Value = "Value3", ValueType = DynamicPropertyValueType.ShortText
+                },
+                new DynamicPropertyObjectValue
+                {
+                    Locale = "fr-FR", Value = "Value4", ValueType = DynamicPropertyValueType.ShortText
+                },
+                new DynamicPropertyObjectValue
+                {
+                    Locale = "de-DE", Value = "Value5", ValueType = DynamicPropertyValueType.ShortText
+                }
+            }).ToArray();
+        }
+
+        [Fact]
+        public async Task Store_And_Usual_Property()
+        {
+            var secondUsualProperty = (DynamicObjectProperty) UsualDynamicProperty.Clone();
+            secondUsualProperty.Name = "Test Property";
+            secondUsualProperty.Values.First().Value = "Test";
+            var properties = new[] { UsualDynamicProperty, secondUsualProperty };
+
+            var store1Customer = new Contact
+            {
+                Id = "Customer1",
+                SecurityAccounts = new[] { new ApplicationUser { StoreId = "Store1" } },
+                DynamicProperties = properties
+            };
+            var store2Customer = new Contact
+            {
+                Id = "Customer2",
+                SecurityAccounts = new[] { new ApplicationUser { StoreId = "Store2" } },
+                DynamicProperties = properties
+            };
+
+            var segment = GetCustomerSegment(new[] { UsualDynamicProperty }, new[] { "Store1" }, "Test");
+
+            var documents = await GetDocuments(segment, store1Customer, store2Customer);
+
+            var firstCustomerUserGroups = GetUserGroups(documents[0]);
+            Assert.Equal("Test", firstCustomerUserGroups.ToString());
+
+            var secondCustomerUserGroups = GetUserGroups(documents[1]);
+            Assert.Null(secondCustomerUserGroups);
+        }
+
+        [Fact]
+        public async Task Multiple_Stores()
```

Comment:
Please fix tests formatting. 
Group it by
// Arrange
// Act
// Assert
Like other VC tests

Reply:
Also tests should be named according microsoft recommendations
https://docs.microsoft.com/ru-ru/dotnet/core/testing/unit-testing-best-practices#naming-your-tests

Reply:
Refactored